samples/javascript_nodejs/70.skills-simple-bot-to-bot/simple-root-bot/index.js

```diff
@@ -0,0 +1,101 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+// index.js is used to setup and configure your bot
+
+// Import required packages
+const path = require('path');
+const restify = require('restify');
+
+// Import required bot services.
+// See https://aka.ms/bot-services to learn more about the different parts of a bot.
+const { BotFrameworkAdapter, ChannelServiceRoutes, ConversationState, InputHints, MemoryStorage, SkillHandler, SkillHttpClient } = require('botbuilder');
+const { AuthenticationConfiguration, SimpleCredentialProvider } = require('botframework-connector');
+
+const { SkillsConfiguration } = require('./skillsConfiguration');
+const { SkillConversationIdFactory } = require('./skillConversationIdFactory');
+
+// This bot's main dialog.
+const { RootBot } = require('./rootBot');
+
+// Import required bot configuration.
+const ENV_FILE = path.join(__dirname, '.env');
+require('dotenv').config({ path: ENV_FILE });
+
+// Create adapter.
+// See https://aka.ms/about-bot-adapter to learn more about adapters.
+const adapter = new BotFrameworkAdapter({
+    appId: process.env.MicrosoftAppId,
```

Comment:
suggestion: it's useful to emit an error if these environment variables aren't present. We see a lot of issues on come to Azure Bot Service related to these not being present and folks getting 401/etc.

Reply:
This is a little more nuanced as undefined/falsy values and empty strings are acceptable parameters here, and in this scenario. Bot-to-bot communication can be "anonymous". I'm going to mark it as out of scope because this would involve changing all the samples in all of the languages.